src/Hosting/Hosting/test/Internal/HostTests.cs

```diff
@@ -709,13 +796,13 @@ public async Task HostStopApplicationDoesNotFireStopOnHostedService()
                 })
                 .Build())
             {
-                var lifetime = host.Services.GetRequiredService<IHostApplicationLifetime>();
-                lifetime.StopApplication();
-
                 await host.StartAsync();
 
                 Assert.Equal(0, stoppingCalls);
                 Assert.Equal(0, disposingCalls);
+
+                var lifetime = host.Services.GetRequiredService<IHostApplicationLifetime>();
+                lifetime.StopApplication();
```

Comment:
Same as below, not sure what the original purpose of this was, but I left it in here to make sure that the `StopApplication` will not call the `StopAsync` of the hosted service.

Reply:
Move above the Asserts?